acq4/devices/MaiTaiLaser/MaiTaiLaser.py

```diff
@@ -0,0 +1,255 @@
+            self.driver.turnLaserOn()
+
+    def switchLaserOff(self):
+        with self.driverLock:
+            self.driver.turnLaserOff()
+            
+    def powerChanged(self, power):
+        with self.maiTaiLock:
+            self.maiTaiPower = power
+            self.setParam(currentPower=power)
+            powerOk = self.checkPowerValidity(power)
+            self.sigOutputPowerChanged.emit(power, powerOk)
+                
+    def wavelengthChanged(self, wl):
+        with self.maiTaiLock:
+            self.maiTaiWavelength = wl
+            self.sigWavelengthChanged.emit(wl)
+    
+    def humidityChanged(self,hum):
+        with self.maiTaiLock:
+            self.maiTaiHumidity = hum
+            self.sigRelativeHumidityChanged.emit(hum)
+            
+    def pumpPowerChanged(self,pP):
+        with self.maiTaiLock:
+            self.maiTaiPumpPower = pP
+            self.sigPumpPowerChanged.emit(pP)
+    
+    def pulsingStateChanged(self, pulse):
+        with self.maiTaiLock:
+            self.maiTaiPumpPower = pulse
+            self.sigPulsingStateChanged.emit(pulse)
+    
+    def humidity(self):
+        with self.maiTaiLock:
+            return self.maiTaiHumidity 
+    
+    def modeChanged(self,mode):
+        with self.maiTaiLock:
+            self.maiTaiMode = mode
+            self.sigModeChanged.emit(mode)
+    
+    def outputPower(self):
+        with self.maiTaiLock:
+            return self.maiTaiPower
+    
+    def getPumpPower(self):
+        with self.maiTaiLock:
+            return self.maiTaiPumpPower
+    
+    def getWavelength(self):
+        with self.driverLock:
+            return self.driver.getWavelength()*1e-9
+        
+    def setWavelength(self, wl):
+        with self.driverLock:
+            self.driver.setWavelength(wl*1e9)
+            
+    def getWavelengthRange(self):
+        with self.driverLock:
+            bounds = self.driver.getWavelengthRange()
+        return bounds[0]*1e-9, bounds[1]*1e-9
+    
+    def openInternalShutter(self):
+        with self.driverLock:
+            self.driver.setShutter(True)
+        
+    def closeInternalShutter(self):
+        with self.driverLock:
+            self.driver.setShutter(False)
+        
+    def getInternalShutter(self):
+        with self.driverLock:
+            return self.driver.getShutter()
+    
+    def externalSwitchON(self):
+        if self.hasExternalSwitch:
+            self.setChanHolding('externalSwitch', 1)
+    
+    def externalSwitchOFF(self):
+        if self.hasExternalSwitch:
+            self.setChanHolding('externalSwitch', 0)
+    
+    def acitvateAlignmentMode(self):
+        with self.driverLock:
+            self.greenPowerIRMode = self.driver.getLastCommandedPumpLaserPower()
+        with self.driverLock:
+            self.driver.setPumpMode('Green Power')
+        self.mThread.alignmentMode = True
+        
+    def deactivateAlignmentMode(self):
+        with self.driverLock:
```

Comment:
Do you need two independent driver locks?

        with self.driverLock:
            self.driver.setPumpLaserPower(self.greenPowerIRMode)
            self.driver.setPumpMode('IR Power')